src/components/MilestoneActions.jsx

```diff
@@ -13,15 +13,72 @@ import ApproveRejectMilestoneCompletionButtons from 'components/ApproveRejectMil
 import WithdrawMilestoneFundsButton from 'components/WithdrawMilestoneFundsButton';
 import EditMilestoneButton from 'components/EditMilestoneButton';
 import ChangeMilestoneRecipientButton from './ChangeMilestoneRecipientButton';
+import { Context as WhiteListContext } from '../contextProviders/WhiteListProvider';
+import { Context as ConversionRateContext } from '../contextProviders/ConversionRateProvider';
+import { Context as UserContext } from '../contextProviders/UserProvider';
 
 function MilestoneActions({ milestone }) {
+  const {
+    actions: { convertMultipleRates },
+  } = useContext(ConversionRateContext);
+  const {
+    state: { minimumPayoutUsdValue },
+  } = useContext(WhiteListContext);
+  const {
+    state: { currentUser },
+  } = useContext(UserContext);
+  const [currentBalanceUsdValue, setCurrentBalanceUsdValue] = useState(0);
+  const [isAmountEnoughForWithdraw, setIsAmountEnoughForWithdraw] = useState(true);
+
+  const calculateMilestoneCurrentBalanceUsdValue = async () => {
+    try {
+      const result = await convertMultipleRates(
+        null,
+        'USD',
+        milestone.donationCounters.map(dc => {
+          return {
+            value: dc.currentBalance,
+            currency: dc.symbol,
+          };
+        }),
+      );
+      setCurrentBalanceUsdValue(result.total);
+    } catch (e) {
+      // TODO should I remove this console.log?
+      console.error('calculateMilestoneCurrentBalanceUsdValue error', e);
+    }
+  };
+
+  useEffect(() => {
+    setIsAmountEnoughForWithdraw(currentBalanceUsdValue > minimumPayoutUsdValue);
+  }, [currentBalanceUsdValue]);
+
+  useEffect(() => {
+    // setLoading(true)
+
+    if (
+      !currentBalanceUsdValue &&
+      currentUser.address &&
+      milestone.donationCounters &&
+      milestone.donationCounters.length
+    ) {
+      calculateMilestoneCurrentBalanceUsdValue();
+    }
+  });
+
   return (
     <Fragment>
       <AcceptRejectProposedMilestoneButtons milestone={milestone} />
 
       <ReproposeRejectedMilestoneButton milestone={milestone} />
 
-      {milestone.hasRecipient ? <RequestMarkMilestoneCompleteButton milestone={milestone} /> : null}
+      {milestone.hasRecipient ? (
+        <RequestMarkMilestoneCompleteButton
+          isAmountEnoughForWithdraw={isAmountEnoughForWithdraw}
+          minimumPayoutUsdValue={minimumPayoutUsdValue}
```

Comment:
It's not necessary to pass it, `RequestMarkMilestoneCompleteButton` can use context itself